### PR TITLE
#67: Increase coverage by unit tests

### PR DESCRIPTION
**Pull request by vitgrams**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by vitgrams** [comment]
@DenTray Done. Could you please review this code?

File: tests/AutoDocMiddlewareTest.php [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+        $this->mockDriverGetEmptyAndSaveTpmData($this->getJsonFixture('tmp_data_search_roles_request'));
+
+        $request = $this->generateRequest('get', 'users/roles', [
+            'with' => ['users']
+        ], [], [
+            'Content-type' => 'application/json'
+        ]);
+
+        $middleware  = new AutoDocMiddleware();
```
Comment (DenTray) [inline tests/AutoDocMiddlewareTest.php:25]:
```suggestion
        $middleware = new AutoDocMiddleware();
```

File: tests/SwaggerServiceTest.php [hunk 2]
```diff
@@ -98,11 +110,51 @@ public function testAddData(string $contentType, string $requestFixture, string
...
+    {
+        $this->mockDriverGetEmptyAndSaveTpmData($this->getJsonFixture('tmp_data_search_roles_request_without_rule_type'));
+
+        $service = app(SwaggerService::class);
+
+        $request = $this->generateRequest('get', 'users/roles', [
+            'with' => ['users']
+        ], [], [
+            'Content-type' => 'application/json'
+        ], 'testRequestWithoutRuleType');
```
Comment (DenTray) [inline tests/SwaggerServiceTest.php:123]: let's move it to the separate `generateGetRolesRequest`

**Review by DenTray — CHANGES_REQUESTED** [review]

**Comment by vitgrams** [comment]
@DenTray All conversations have benn resolved. Could you please take another look at this?

**Review by DenTray — CHANGES_REQUESTED** [review]

**Review by DenTray — CHANGES_REQUESTED** [review]

File: tests/TestCase.php [hunk 3]
```diff
@@ -61,17 +107,37 @@ protected function clearDirectory($dirPath, $exceptPaths = [])
...
+    protected function generateGetRolesRequest($method = 'test'): Request
+    {
+        return $this->generateRequest('get', 'users/roles', [
+            'with' => ['users']
+        ], [], [
+            'Content-type' => 'application/json'
+        ], $method);
+    }
+
+    protected function generateResponse($fixture, int $status = 200, array  $headers = []): Response
```
Comment (DenTray) [inline tests/TestCase.php:130]:
```suggestion
    protected function generateResponse($fixture, int $status = 200, array $headers = []): Response
```

File: tests/TestCase.php [hunk 4]
```diff
@@ -61,17 +107,37 @@ protected function clearDirectory($dirPath, $exceptPaths = [])
...
+    {
+        if (empty($headers)) {
+            $headers = [
+                'Content-type' => 'application/json',
+                'authorization' => 'Bearer some_token'
+            ];
+        }
+        return new Response(($fixture) ? $this->getFixture($fixture) : null, $status, $headers);
```
Comment (DenTray) [inline tests/TestCase.php:138]:
```suggestion

        return new Response(($fixture) ? $this->getFixture($fixture) : null, $status, $headers);
```

**Review by DenTray — CHANGES_REQUESTED** [review]

**Review by DenTray — APPROVED** [review]

**Pull request merged**